mlflow/rfunc/backend.py

```diff
@@ -0,0 +1,69 @@
+import logging
+import os
+import re
+import subprocess
+
+from mlflow.models import FlavorBackend
+from mlflow.tracking.artifact_utils import _download_artifact_from_uri
+from mlflow.utils.file_utils import TempDir
+
+_logger = logging.getLogger(__name__)
+
+
+class RFuncBackend(FlavorBackend):
+    """
+    Flavor backend implementation for the generic R models.
+    Predict and serve locally models with 'crate' flavor.
+    """
+    version_pattern = re.compile("version ([0-9]+[.][0-9]+[.][0-9]+)")
+
+    def predict(self, model_uri, input_path, output_path, content_type, json_format):
+        """
+        Generate predictions using R model saved with MLflow.
+        Return the prediction results as a JSON.
+        """
+        with TempDir() as tmp:
+            model_path = _download_artifact_from_uri(model_uri, output_path=tmp.path())
+            str_cmd = "mlflow:::mlflow_rfunc_predict('{0}', {1}, {2}, {3})"
```

Comment:
We could escape `model_path` to be extra safe here if the model path contains e.g. quotes (apparently quotes are valid characters within e.g. unix paths).

Reply:
Good point. But I also think it's ok to do this later.